preprocessing-service/preprocess.py

```diff
@@ -36,32 +76,39 @@ async def mask_logs(queue):
         pending_list.append(payload)
         start_time = time.time()
         if start_time - last_time >= 1 or len(pending_list) >= 128:
-            payload_data_df = pd.DataFrame(pending_list)
+            payload_list = PayloadList(items=pending_list)
             last_time = start_time
             pending_list = []
-            await run(payload_data_df, masker)
-
-async def run(payload_data_df, masker):
-    logging.info(f"processing {len(payload_data_df)} logs...")
-    payload_data_df["log"] = payload_data_df["log"].str.strip()
-    masked_logs = []
-
-    for index, row in payload_data_df.iterrows():
+            await run(payload_list, masker)
+
+async def run(payload_list, masker):
+    logging.info(f"processing {len(payload_list.items)} logs...")
+    start_time = time.time()
+    filtered_workload_logs = []
+    filtered_pretrained_logs = []
+    for payload in payload_list.items:
         try:
-            masked_logs.append(masker.mask(row["log"]))
+            if payload.log_type == "workload":
+                if verify_workload(payload):
+                    payload.masked_log = masker.mask(payload.log)
+                    filtered_workload_logs.append(payload)
+            else:
+                payload.masked_log = masker.mask(payload.log)
+                filtered_pretrained_logs.append(payload)
         except Exception as e:
-            masked_logs.append(row["log"])
```

Comment:
we should keep this line --> if masker failed for some reason we add the raw logs as masked logs?

Reply:
Discussed in-person